WordPress/Classes/Stores/StatsInsightsStore.swift

```diff
@@ -888,22 +894,22 @@ extension StatsInsightsStore {
                 return []
         }
 
-        let postingEvents = state.postingActivity?.postingEvents ?? []
-
         // This gives a range of how many days there are in a given month...
         let rangeOfMonth = calendar.range(of: .day, in: .month, for: date) ?? 0..<0
 
         let mappedMonth = rangeOfMonth
             // then we create a `Date` representing each of those days
-            .compactMap {
-                return calendar.date(from: DateComponents(year: year, month: month, day: $0))
-            }
             // and pick out a relevant `PostingStreakEvent` from data we have or return
             // an empty one.
-            .map { (date: Date) -> PostingStreakEvent in
-                if let postingEvent = postingEvents.first(where: { event in return event.date == date }) {
+            .compactMap { (day: Int) -> PostingStreakEvent? in
+                guard let date = calendar.date(from: DateComponents(year: year, month: month, day: day)) else {
```

Comment:
(just a note) I noticed a lot of these date manipulations in the stats. It's easy to make a mistake. It can be simplified and optimized by using a so-called [naive date type](https://github.com/CreateAPI/NaiveDate) that has no timezones.

Reply:
Interesting, thanks for sharing!